torch/csrc/jit/profiling_record.cpp

```diff
@@ -0,0 +1,118 @@
+#include <jit/profiling_record.h>
+
+namespace torch {
+namespace jit {
+
+void ProfilingRecord::addProfilingRecordAttribute(Node* pop) {
+  int64_t addr = 0; // if sizeof(ptr) == 4
+  addr = reinterpret_cast<int64_t>(this);
+  pop->i_(attr::storage, addr);
+}
+
+Node* ProfilingRecord::createProfileNode(Graph* graph, Value* o) {
+  Node* pop = graph->create(prim::profile, 0);
+  pop->i_(attr::data, reinterpret_cast<int64_t>(o));
+  addProfilingRecordAttribute(pop);
+  return pop;
+}
+
+void ProfilingRecord::instrumentBlock(Block* block) {
+  auto graph = block->owningGraph();
+  for (auto it = block->nodes().rend(); it != block->nodes().rbegin(); --it) {
```

Comment:
Do we go backwards to avoid instrumenting `::profile` nodes? Please add a comment.

Reply:
added